huaweicloud/services/ecs/resource_huaweicloud_compute_volume_attach.go

```diff
@@ -87,53 +90,53 @@ func resourceComputeVolumeAttachCreate(d *schema.ResourceData, meta interface{})
 		ServerId: instanceId,
 	}
 
-	logp.Printf("[DEBUG] Creating volume attachment: %#v", attachOpts)
+	log.Printf("[DEBUG] Creating volume attachment: %#v", attachOpts)
 	job, err := block_devices.Attach(computeClient, attachOpts)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
-	logp.Printf("[DEBUG] The response of volume attachment request is: %#v", job)
+	log.Printf("[DEBUG] The response of volume attachment request is: %#v", job)
 	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"INIT", "RUNNING"},
-		Target:     []string{"SUCCESS"},
-		Refresh:    AttachmentJobRefreshFunc(computeClient, job.ID),
-		Timeout:    d.Timeout(schema.TimeoutCreate),
-		Delay:      30 * time.Second,
-		MinTimeout: 15 * time.Second,
+		Pending:      []string{"INIT", "RUNNING"},
+		Target:       []string{"SUCCESS"},
+		Refresh:      AttachmentJobRefreshFunc(computeClient, job.ID),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        30 * time.Second,
+		PollInterval: 15 * time.Second,
 	}
 
-	if _, err = stateConf.WaitForState(); err != nil {
-		return fmtp.Errorf("Error attaching HuaweiCloud volume: %s", err)
+	if _, err = stateConf.WaitForStateContext(ctx); err != nil {
+		return diag.Errorf("Error attaching volume: %s", err)
 	}
 
 	// The old logic use the instance ID and attachment ID as the resource ID, and the attachment ID equals
 	// the volume ID.
 	id := fmt.Sprintf("%s/%s", instanceId, volumeId)
 	d.SetId(id)
 
-	return resourceComputeVolumeAttachRead(d, meta)
+	return resourceComputeVolumeAttachRead(ctx, d, meta)
 }
 
-func resourceComputeVolumeAttachRead(d *schema.ResourceData, meta interface{}) error {
+func resourceComputeVolumeAttachRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	region := GetRegion(d, config)
+	region := config.GetRegion(d)
 	computeClient, err := config.ComputeV1Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud compute client: %s", err)
+		return diag.Errorf("Error creating compute V1 client: %s", err)
 	}
 
-	instanceId, volumeId, err := parseComputeVolumeAttachmentId(d.Id())
+	instanceId, volumeId, err := ParseComputeVolumeAttachmentId(d.Id())
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	attachment, err := block_devices.Get(computeClient, instanceId, volumeId).Extract()
 	if err != nil {
-		return CheckDeleted(d, err, "compute_volume_attach")
+		return common.CheckDeletedDiag(d, err, "compute_volume_attach")
```

Comment:
suggest to add a more detailes message to keep the error style same

Reply:
done